extensions/sql-migration/src/wizard/loginSelectorPage.ts

```diff
@@ -396,6 +399,8 @@ export class LoginSelectorPage extends MigrationWizardPage {
 				text: constants.LOGIN_MIGRATIONS_GET_LOGINS_ERROR_TITLE('target'),
 				description: constants.LOGIN_MIGRATIONS_GET_LOGINS_ERROR(error.message),
 			};
+
+			logError(TelemetryViews.LoginMigrationWizard, 'CollectingTargetLoginsFailed', error);
```

Comment:
TelemetryReporter.sendErrorEvent takes in the error parameter and expects it to be a string. But you aren't guaranteeing that (and in fact most errors will be Error objects) so this isn't going to work as expected.

Reply:
This is the pattern we use throughout sql-migraiton. I think it should be fine.  [logError function](https://github.com/microsoft/azuredatastudio/blob/main/extensions/sql-migration/src/telemetry.ts#L74) wraps the Telemetry.sendErrorEvent as such:

```
export function logError(telemetryView: TelemetryViews, err: string, error: any): void {
	console.log(error);
	TelemetryReporter.sendErrorEvent(telemetryView, err);
}
```

so the param passed to Telemetry.sendErrorEvent will be a string, and we are emitting the error object to console.log

Reply:
But you're just passing it the error directly in this code. It doesn't complain because the type is any at that point (default type for errors in catch). So error could be an object for all you know here and it would compile fine then just not work correctly at runtime. 

[Example](https://www.typescriptlang.org/play?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCbvCwDKgU8JkY7p7ehCTkVDQS2E6gnPCxGcwmZqDSTgzxxWWVoASMFmgYkAAeRJTInN3ymj4d-jSCeNsMq-wuoPaOltigAKoASgAywhK7SbGQZIIz5VWCFzSeCrZagNYbChbHaxUDcCjJZLfSDbExIAgUdxkUBIursJzCFJtXydajBBSCVAACgA5NhoNB4DSAJQAbkUFOpAG95mwCPlQHSYIyaaAAL5iYRmPBsjn6bm8+D8wX0kXiyWgLjxAYSTiy5JcWD1RpoeVUylkAUW4bMgU6eAUVxc4Ji4LFHDNDxkLyIeB9BjA9g0L5uMzMdgMZiQfiwaDqRIB+B6EReABEAFFVgxEAt9KmktjU+AHKgacIAHKQCSpoIqADyXzI70E0a0kCOqEcClc0jwoCcvp0wyw+hQCEQ0KlPBHkj90REBg42IWTFYHB4QRKQA)

You should be able to see this if you set the [useUnknownInCatchVariables](https://www.typescriptlang.org/tsconfig#useUnknownInCatchVariables) compiler flag to true in your tsconfig.json. (which is probably something we should enable for all our extensions at some point)

Reply:
hmm I guess I'm not seeing the issue here. the param `error` isn't being be passed to Telemetry.sendErrorEvent, and it's only being used in console.log. Even if it is an object, won't the object get logged without issues. 


again to clarify, the second param `err` is a string and that is the param passed to Telemetry.sendErrorEvent. The third param `error`, which is the one that is caught will be passed in as any (so no compiler issue) and is only passed into console.log.

Reply:
And this is just me not actually reading the code correctly - you're completely right. So nevermind then - apologies for the random tangent!